war/src/main/java/io/lumeer/engine/controller/ViewFacade.java

```diff
@@ -310,102 +286,103 @@ public Object getViewMetadataValue(int viewId, String metaKey) throws ViewMetada
-      setViewMetadataValue(viewDocument, LumeerConst.Document.USER_RIGHTS, viewDocument.getDataDocument(LumeerConst.Document.USER_RIGHTS));
+   public void updateViewAccessRights(final DataDocument viewDocument) {
+      setViewMetadataValue(viewDocument.getInteger(LumeerConst.View.ID_KEY), LumeerConst.Document.USER_RIGHTS, viewDocument.getDataDocument(LumeerConst.Document.USER_RIGHTS));
    }
 
-   // sets info about one view without checking special metadata keys
-   private void setViewMetadataValue(DataDocument viewDocument, String metaKey, Object value) throws ViewMetadataNotFoundException, UnauthorizedAccessException {
-      if (!securityFacade.checkForWrite(viewDocument, getCurrentUser())) {
-         throw new UnauthorizedAccessException();
-      }
+   public boolean checkViewForRead(int viewId, String user) {
+      return securityFacade.checkForRead(getViewMetadataDocument(viewId), user);
+   }
+
+   public boolean checkViewForWrite(int viewId, String user) {
+      return securityFacade.checkForWrite(getViewMetadataDocument(viewId), user);
+   }
 
-      DataDocument metadataDocument = new DataDocument(metaKey, value)
-            .append(LumeerConst.View.VIEW_UPDATE_USER_KEY, getCurrentUser())    // with every change, we change update user and date
-            .append(LumeerConst.View.VIEW_UPDATE_DATE_KEY, new Date());
+   public boolean checkViewForAccessChange(int viewId, String user) {
+      return securityFacade.checkForExecute(getViewMetadataDocument(viewId), user);
+   }
+
+   private void setViewMetadataValue(int viewId, String metaKey, Object value) {
+      DataDocument metadataDocument = new DataDocument(metaKey, value);
+      setUpdateTimeAndUser(metadataDocument);
+
+      dataStorage.updateDocument(
+            metadataCollection(),
+            metadataDocument,
+            dataStorageDialect.fieldValueFilter(LumeerConst.View.ID_KEY, viewId));
+   }
 
-      String id = viewDocument.getId();
-      dataStorage.updateDocument(LumeerConst.View.VIEW_METADATA_COLLECTION_NAME, metadataDocument, dataStorageDialect.documentIdFilter(id));
+   private DataDocument setUpdateTimeAndUser(DataDocument updatedDocument) {
+      return updatedDocument
+            .append(LumeerConst.View.UPDATE_USER_KEY, getCurrentUser())    // with every change, we change update user and date
+            .append(LumeerConst.View.UPDATE_DATE_KEY, new Date());
    }
 
    private String getCurrentUser() {
       return userFacade.getUserEmail();
    }
+
+   /**
+    * @return name of view metadata collection for current project
+    */
+   public String metadataCollection() {
+      return metadataCollection(projectFacade.getCurrentProjectId());
+   }
+
+   /**
+    * @param projectId
+    *       project id
+    * @return name of view metadata collection for given project id
+    */
+   private String metadataCollection(String projectId) {
+      return LumeerConst.View.METADATA_COLLECTION_PREFIX + projectId;
+   }
```

Comment:
You should use also organizationId to create collection. Or make separate collection for each organization and then use projectId as one of the field (don't forget to add projectId to index then)...